src/test/scala/advancepassengerinfo/importer/processor/DqFileProcessorImplTest.scala

```diff
@@ -209,6 +210,68 @@ class DqFileProcessorTest extends TestKit(ActorSystem("MySpec"))
       probe.expectMsg(ManifestCall(jsonFileName, manifest))
       probe.expectMsg(JsonFileCall("c", jsonFileName, successful = true, dateIsSuspicious = false))
       probe.expectMsg(ZipFileCall("c", successful = true))
+
+    }
+
+    "handle exceptions while getting nextFiles" in {
+      val mockFileNamesProvider = new FileNames {
+        override val nextFiles: String => Future[List[String]] = (previous: String) => Future.sequence(List(
+          Future.successful(List("1.zip", "2.zip")),
+          Future.failed(new Exception("next file exception")),
+          Future.successful(List("3.zip", "4.zip")),
+        )).map(_.flatten)
+      }
+
+      val probe = TestProbe("probe")
+      val mockPersistence = MockPersistence(probe.ref)
+
+      val manifests = Seq(
+        ("1.json", Success(manifest)),
+        ("2.json", Success(manifest)),
+      )
+      val processor = DqFileProcessorImpl(singleZipMockProvider(manifests), mockPersistence)
+      val dqApiFeed: DqApiFeedImpl = DqApiFeedImpl(
+        mockFileNamesProvider,
+        processor,
+        100.millis,
+        MockStatsDCollector
+      )
+
+      val result = Await.result(dqApiFeed.processFilesAfter("_").runWith(Sink.seq),1.second)
+
+      result should ===(Vector.empty)
+    }
+
+    "handle exception while persisting" in {
+      val mockPersistence = new Persistence {
+        override def persistManifest(jsonFileName: String, manifest: VoyageManifest): Future[Option[Int]] = Future.failed(new Exception("db exception"))
+
+        override def persistJsonFile(zipFileName: String, jsonFileName: String, successful: Boolean, dateIsSuspicious: Boolean): Future[Int] = Future.failed(new Exception("db exception"))
+
+        override def persistZipFile(zipFileName: String, successful: Boolean): Future[Boolean] = Future.failed(new Exception("db exception"))
+
+        override def lastPersistedFileName: Future[Option[String]] = Future.failed(new Exception("db exception"))
+
+        override def jsonHasBeenProcessed(zipFileName: String, jsonFileName: String): Future[Boolean] = Future.failed(new Exception("db exception"))
+      }
+
+      val manifests = Seq(
+        ("1.json", Success(manifest)),
+        ("2.json", Success(manifest)),
+        ("3.json", Success(manifest)),
+        ("4.json", Success(manifest)),
+      )
+      val processor = DqFileProcessorImpl(singleZipMockProvider(manifests), mockPersistence)
+      val dqApiFeed: DqApiFeedImpl = DqApiFeedImpl(
+        MockFileNames(List(List("1.zip", "2.zip"), List("3.zip", "4.zip"))),
+        processor,
+        100.millis,
+        MockStatsDCollector
+      )
+
+      val result = Await.result(dqApiFeed.processFilesAfter("_").runWith(Sink.seq),1.second)
+
+      result should ===(Vector.empty)
```

Comment:
I don't think this is a useful test. What does it show? We need to see that the app continues to process files after a failure